### DISCUSSION TITLE
umi@4 怎样在组件外获取model数据流

### DISCUSSION THREAD

**Discussion by sunyongming** [umijs/umi#10141] [Q&A] [answered]
umi@4 怎样在组件外获取model数据流，非组件没法用useModel，想在请求拦截器里获取model

**Comment by fz6m** [comment, marked as answer]
要脱离 react hooks 的数据操作，需要放到全局状态管理库里，比如 [valtio](https://umijs.org/docs/max/valtio) 。

**Reply by sunyongming** [reply]
只是不太优雅，layout和access都可以共享[initialState](https://umijs.org/docs/max/data-flow#%E5%85%A8%E5%B1%80%E5%88%9D%E5%A7%8B%E7%8A%B6%E6%80%81)，request也能共享initialState就好了

**Comment by AtLastDopamine** [comment]
终于有人问这个问题了 🫠

我之前从 `Vue` 转过来用 `UMI` 的，习惯 `VueX`、`Pinia`，然后自然认为 `model` 也支持在非 `React` 上下文中读写。看大家都没有这个问题，我还以为是有什么 Api 直接使用，只是自己不知道。疯狂 Google 冲浪 🏄🏻

太真实了，小丑竟是我自己  🤡
